app/src/main/java/org/asteroidos/sync/asteroid/AsteroidBleManager.java

```diff
@@ -0,0 +1,148 @@
+        return new AsteroidBleManagerGattCallback();
+    }
+
+    public final void abort() {
+        cancelQueue();
+    }
+
+    @Override
+    protected final void finalize() throws Throwable {
+        super.finalize();
+    }
+
+    public final void setBatteryLevel(Data data) {
+        BatteryLevelEvent batteryLevelEvent = new BatteryLevelEvent();
+        batteryLevelEvent.battery = Objects.requireNonNull(data.getByte(0)).intValue();
+        synchronizationService.handleBattery(batteryLevelEvent);
+    }
+
+    public static class BatteryLevelEvent {
+        public int battery = 0;
+    }
+
+    private class AsteroidBleManagerGattCallback extends BleManagerGattCallback {
+
+        @Override
+        public final boolean isRequiredServiceSupported(@NonNull final BluetoothGatt gatt) {
+            final BluetoothGattService batteryService = gatt.getService(AsteroidUUIDS.BATTERY_SERVICE_UUID);
+
+            boolean supported = true;
+
+            boolean notify = false;
+            if (batteryService != null) {
+                batteryCharacteristic = batteryService.getCharacteristic(AsteroidUUIDS.BATTERY_UUID);
+
+                if (batteryCharacteristic != null) {
+                    final int properties = batteryCharacteristic.getProperties();
+                    notify = (properties & BluetoothGattCharacteristic.PROPERTY_NOTIFY) != 0;
+                }
+            }
+            if(sendingCharacteristics == null)
+                sendingCharacteristics = new HashMap<>();
+
+            for (IConnectivityService service : synchronizationService.getServices()) {
+
+                BluetoothGattService bluetoothGattService = gatt.getService(service.getServiceUUID());
+                List<UUID> sendUuids = new ArrayList<>();
+                service.getCharacteristicUUIDs().forEach((uuid, direction) -> {
+                    if (direction == IConnectivityService.Direction.TO_WATCH)
+                        sendUuids.add(uuid);
+                });
+                Log.d(TAG, "UUID " + sendUuids);
+
+                for (UUID uuid: sendUuids) {
+                    BluetoothGattCharacteristic characteristic = bluetoothGattService.getCharacteristic(uuid);
+                    sendingCharacteristics.put(uuid, characteristic);
+                    bluetoothGattService.addCharacteristic(characteristic);
+                }
+                recvCallbacks.forEach((characteristic, callback) -> {
+                    BluetoothGattCharacteristic characteristic1 = bluetoothGattService.getCharacteristic(characteristic);
+                    setNotificationCallback(characteristic1).with((device, data) -> callback.call(data.getValue()));
+                    enableNotifications(characteristic1).with((device, data) -> callback.call(data.getValue())).enqueue();
+                });
+            }
+
+            supported = (batteryCharacteristic != null && notify);
+
+            // Return true if all required services have been found
```

Comment:
Is this comment true ? it seems to me that you are only checking for the presence of the battery service ?